subprojects/gradle/namespaced-resources-fixer/src/main/kotlin/com/avito/android/plugin/FileResourcesFixer.kt

```diff
@@ -0,0 +1,169 @@
+package com.avito.android.plugin
+
+import io.github.detekt.parser.KtCompiler
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.com.intellij.psi.PsiWhiteSpace
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtImportList
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+import java.io.File
+
+/**
+ * Replace unambiguous R class references to Application by libraries R classes
+ */
+internal class FileResourcesFixer {
+
+    private val compiler = KtCompiler()
+
+    /**
+     * @return new file content or null if nothing changed
+     */
+    fun fixMergedResources(file: File, app: AndroidModule, libraries: List<AndroidModule>): String? {
+        val ktFile = compiler.compile(file.toPath(), file.toPath())
+
+        val mergedRClassImport = findMergedRImport(ktFile, app) ?: return null
+
+        val mergedRClassReferencesFinder = MergedRClassReferencesFinder(mergedRClassImport)
+        mergedRClassReferencesFinder.visitKtFile(ktFile)
+
+        val references = mergedRClassReferencesFinder.references
+        if (references.isEmpty()) return null
+
+        val uniqueResources: List<ResourceId> = findUniqueResources(libraries) // TODO: support string, drawable, ...
+
+        val replacedModules = replaceReferences(references, uniqueResources)
+        fixImports(ktFile, replacedModules)
+
+        return ktFile.text
+    }
+
+    private fun replaceReferences(
+        rClassReferences: List<PsiElement>,
+        uniqueResources: List<ResourceId>
+    ): Set<AndroidModule> {
+        val replacedModules = mutableSetOf<AndroidModule>()
+        rClassReferences.forEach { refElement ->
+            val resource = uniqueResources.find { resource ->
+                refElement.text.endsWith(".id.${resource.id}")
```

Comment:
It will work incorrect with identical ids from different modules
```
import com.android.app.R as app_R
import com.android.library.R as library_R

val r = app_R.id.one
val r1 = library_R.id.one
```

Reply:
Added new test `replace library id - two simple properties with the same name`